Earthfile

```diff
@@ -455,79 +436,149 @@ uki-tools-image:
+        --secureboot-certificate DB.crt \
+        --pcr-private-key private.pem \
+        --measure \
+        --output uki.signed.efi
     RUN sbsign --key DB.key --cert DB.crt --output systemd-bootx64.signed.efi /usr/lib/systemd/boot/efi/systemd-bootx64.efi
-    RUN sbsign --key DB.key --cert DB.crt --output uki.signed.efi uki.unsigned.efi
+    RUN printf 'title Kairos %s\nefi /EFI/kairos/kairos-%s.efi\nversion %s' ${KAIROS_VERSION} ${KAIROS_VERSION} ${KAIROS_VERSION} > kairos-${KAIROS_VERSION}.conf
+    RUN printf 'default @saved\ntimeout 5\nconsole-mode max\neditor no\n' > loader.conf
     SAVE ARTIFACT PK.der PK.der
     SAVE ARTIFACT KEK.der KEK.der
     SAVE ARTIFACT DB.der DB.der
-    SAVE ARTIFACT systemd-bootx64.signed.efi systemd-bootx64.efi
+    SAVE ARTIFACT systemd-bootx64.signed.efi systemd-bootx64.signed.efi
     SAVE ARTIFACT uki.signed.efi uki.signed.efi
-    SAVE ARTIFACT uki.unsigned.efi uki.unsigned.efi
+    SAVE ARTIFACT kairos-${KAIROS_VERSION}.conf kairos-${KAIROS_VERSION}.conf
+    SAVE ARTIFACT loader.conf loader.conf
 
-# Copy uki artifacts into local build dir
-uki-local-artifacts:
-    FROM +uki
+# Base target to set the directory structure for the image artifacts
+# as we need to create several dirs and copy files into them
+# Then we generate the image from scratch to not ring anything else
+uki-image-artifacts:
+    FROM +uki-tools-image
     COPY +version/VERSION ./
-    ARG VERSION=$(cat VERSION)
-    COPY +uki/systemd-bootx64.efi systemd-bootx64.efi
-    COPY +uki/uki.signed.efi uki.signed.efi
-    RUN printf "title Kairos ${FLAVOR} ${VERSION}\nefi /EFI/kairos/kairos.efi" > kairos.conf
-    RUN printf "default kairos.conf" > loader.conf
-    SAVE ARTIFACT systemd-bootx64.efi systemd-bootx64.efi AS LOCAL build/systemd-bootx64.efi
-    SAVE ARTIFACT uki.signed.efi uki.signed.efi AS LOCAL build/uki.${FLAVOR}.${VERSION}.efi
-    SAVE ARTIFACT kairos.conf kairos.conf AS LOCAL build/kairos.conf
-    SAVE ARTIFACT loader.conf loader.conf AS LOCAL build/loader.conf
+    RUN echo "version ${VERSION}"
+    ARG KAIROS_VERSION=$(cat VERSION)
+    COPY +uki-build/systemd-bootx64.signed.efi /output/efi/EFI/BOOT/BOOTX64.EFI
+    COPY +uki-build/uki.signed.efi /output/efi/EFI/kairos/kairos-${KAIROS_VERSION}.efi
+    COPY +uki-build/kairos-${KAIROS_VERSION}.conf /output/efi/loader/entries/kairos-${KAIROS_VERSION}.conf
+    COPY +uki-build/loader.conf /output/efi/loader/loader.conf
+    COPY +uki-build/PK.der /output/efi/loader/keys/kairos/PK.der
+    COPY +uki-build/KEK.der /output/efi/loader/keys/kairos/KEK.der
+    COPY +uki-build/DB.der /output/efi/loader/keys/kairos/DB.der
+    SAVE ARTIFACT /output/efi efi
+
+# This is the final artifact, only the files on it
+uki-image:
+    ARG --required FAMILY
+    COPY +version/VERSION ./
+    ARG KAIROS_VERSION=$(cat VERSION)
+    FROM scratch
+    COPY +uki-image-artifacts/efi /
+    SAVE IMAGE --push $IMAGE_REPOSITORY_ORG/${FAMILY}:${KAIROS_VERSION}-uki
```

Comment:
yup as discussed you should be able to compose the new name with `naming.sh`

```
    ARG OS_NAME=kairos-${VARIANT}-${FLAVOR}-${FLAVOR_RELEASE}
    RUN KAIROS_VERSION=$(cat ./VERSION) \
        OS_VERSION=$(cat ./VERSION) \
        OS_LABEL=$(cat ./VERSION) \
        OS_LABEL=$(naming.sh container_artifact_label) \
        OS_REPO=$(naming.sh container_artifact_repo) \
        ARTIFACT=$(naming.sh bootable_artifact_name) \
        envsubst >>/etc/os-release </usr/lib/os-release.tmpl
    RUN KAIROS_VERSION=$(cat ./VERSION) naming.sh container_artifact_name > /IMAGE

    ARG _CIMG=$(cat ./IMAGE)
```

Reply:
we used to have a command for this, but since for a little while it was done in the dockerfile, I removed it, maybe I should bring it back, but we can do that after this PR

Reply:
We will soon replace the naming.sh script with versioneer (though the standalone cli or kairos-agent). If you find `naming.sh` needing any changes, let's coordinate first.

Reply:
yes indeed, we could have the uki as part of the name maybe in the versioneer? no idea where, if as the model or something (so generic and uki) ?